app/preload/database/transactionsDao.ts

```diff
@@ -89,7 +89,14 @@ export async function insertTransactions(
     transactions: Partial<TransferTransaction>[]
 ) {
     const table = (await knex())(transactionTable);
-    const existingTransactions: TransferTransaction[] = await table.select();
+
+    const hashes = transactions
+        .map((t) => t.transactionHash || '')
```

Comment:
Can you flip the order of `map` and `filter` here? So it becomes:
```
.filter((t) => t.transactionHash).map(t.transactionHash);
```
That way we can perhaps get rid of the `|| ''`.